src/main/java/seedu/mtracker/commands/AddEtfCommand.java

```diff
@@ -1,11 +1,41 @@
 package seedu.mtracker.commands;
 
+import seedu.mtracker.instrument.subinstrument.Etf;
+import seedu.mtracker.ui.TextUi;
+
 public class AddEtfCommand extends AddInstrumentCommand {
 
     public static final String COMMAND_WORD = "etf";
 
+    public static final int PAST_RETURNS_INDEX = 3;
+    public static final int REMARK_INDEX = 4;
+
+    private static final double UNDEFINED_PAST_RETURN_VALUE = 0;
+
+    protected double pastReturnsParameter;
+    protected String remarkParameter;
+
+    public void setEtfParameters() {
+        remarkParameter = inputParameters.get(REMARK_INDEX);
+        try {
+            pastReturnsParameter = Double.parseDouble(inputParameters.get(PAST_RETURNS_INDEX));
+        } catch (NumberFormatException e) {
+            pastReturnsParameter = UNDEFINED_PAST_RETURN_VALUE;
+        }

```

Comment:
Shift this validation and undefined_past_return_value into add instrument Parser. Make undefined_past_return_value to -101 as theoretical all valid range of values range from -100 to infinity